src/widgets/table/table.rs

```diff
@@ -564,8 +566,39 @@ impl<'a> Table<'a> {
         reason = "The name for this feature is not final and may change in the future",
         issue = "https://github.com/ratatui-org/ratatui/issues/536"
     )]
-    pub const fn segment_size(mut self, segment_size: SegmentSize) -> Self {
-        self.segment_size = segment_size;
+    #[deprecated(note = "please use flex instead of segment_size")]
+    pub const fn segment_size(self, segment_size: SegmentSize) -> Self {
+        let translated_to_flex = match segment_size {
+            SegmentSize::None => Flex::Start,
+            SegmentSize::EvenDistribution => Flex::Stretch,
+            SegmentSize::LastTakesRemainder => Flex::StretchLast,
+        };
+
+        self.flex(translated_to_flex)
+    }
+
+    /// Set how extra space is distributed amongst columns.
+    ///
+    /// This determines how the space is distributed when the constraints are satisfied. By default,
+    /// the extra space is not distributed at all.  But this can be changed to distribute all extra
+    /// space to the last column or to distribute it equally.
+    ///
+    /// This is a fluent setter method which must be chained or used as it consumes self
+    ///
+    /// # Examples
+    ///
+    /// Create a table that needs at least 30 columns to display.  Any extra space will be assigned
+    /// to the last column.
+    #[cfg_attr(feature = "unstable", doc = " ```")]
+    #[cfg_attr(not(feature = "unstable"), doc = " ```ignore")]
```

Comment:
I'm curious, why is this required?

Reply:
I don't know. I copy pasted this part from the Table::segment_size's documentation.

Reply:
Do the tests pass without it?

Reply:
For `Table::flex` yes, but `Table::segment_size` fails. I think we can remove it from `Table::flex`.

```
---- src/widgets/table/table.rs - widgets::table::table::Table<'a>::segment_size (line 556) stdout ----
error[E0603]: enum `SegmentSize` is private
  --> src/widgets/table/table.rs:558:35
   |
5  | use ratatui::layout::{Constraint, SegmentSize};
   |                                   ^^^^^^^^^^^ private enum
   |
note: the enum `SegmentSize` is defined here
  --> /Users/emivvvvv/Documents/GitHub/ratatui/src/layout.rs:24:16
   |
24 | pub(crate) use segment_size::SegmentSize;
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0624]: method `segment_size` is private
   --> src/widgets/table/table.rs:562:6
    |
9   |       .segment_size(SegmentSize::LastTakesRemainder);
    |        ^^^^^^^^^^^^ private method
    |
   ::: /Users/emivvvvv/Documents/GitHub/ratatui/src/widgets/table/table.rs:563:5
    |
563 | /     #[stability::unstable(
564 | |         feature = "segment-size",
565 | |         reason = "The name for this feature is not final and may change in the future",
566 | |         issue = "https://github.com/ratatui-org/ratatui/issues/536"
567 | |     )]
    | |______- private method defined here
    ```

Reply:
Yes it should be removed from `Table::flex` and kept for `segment_size`. That's to indicate this is subject to change, `flex` isn't really.